stock_measuring_device/components/measuring_device_component.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import logging
+
+from odoo.addons.component.core import AbstractComponent
+
+_logger = logging.getLogger(__name__)
+
+
+class MeasuringDevice(AbstractComponent):
+    _name = "measuring.device.base"
+    _collection = "measuring.device"
+
+    def test_device(self):
+        success = self._do_test_device()
+        device = self.collection
+        if success and device.state == "not_ready":
+            device.state = "ready"
+        elif not success and device.state == "ready":
+            device.state = "not_ready"
+
+    def _do_test_device(self):
```

Comment:
I guess this has to be overridden by specific implementation?
A docstring would help :wink: